### PR TITLE
Guide explaining how to keep compatibility with mkdocs

### PR DESCRIPTION
**Pull request by humitos**
Guide to help users that weren't pinning a version of `mkdocs` but were relying on `mkdocs==0.15`. Since we upgrade our default version installed, they will find that their docs are not building anymore.

This guide helps them to keep building their docs on Read the Docs showing the workarounds needed.

I'm not really happy with the document and how it's written but I think it communicates what we need.

Related to #4041 and #4556

### CODE REVIEW

File: docs/guides/mkdocs-old-versions.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
+Keep building docs with old version of MkDocs
```
Comment (stsewd) [inline docs/guides/mkdocs-old-versions.rst]: We use title case for titles

File: docs/guides/mkdocs-old-versions.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+=============================================
+
+Recent changes on ``mkdocs`` forced us to `upgrade the default version installed`_ by Read the Docs and this may be a breaking change for your documentation.
+
+.. _upgrade the default version installed: https://github.com/rtfd/readthedocs.org/pull/4041
+
+You should check that your docs continue building in any of these cases:
+
+* your project doesn't have a ``requirements.txt`` file pinning ``mkdocs`` to an specific version
```
Comment (stsewd) [inline docs/guides/mkdocs-old-versions.rst]: `an` -> `a`

**Comment by humitos** [comment]
@stsewd thanks! I took your comments.

**Comment by humitos** [comment]
@davidfischer please, when you have some time, take a look at this PR since you were helping me to understand the underlying problem and find the solution.

File: docs/guides/mkdocs-old-versions.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
+Keep Building Docs With Old Version Of MkDocs
+=============================================
+
+Recent changes on ``mkdocs`` forced us to `upgrade the default version installed`_ by Read the Docs and this may be a breaking change for your documentation.
```
Comment (davidfischer) [inline docs/guides/mkdocs-old-versions.rst]: on -> to

File: docs/guides/mkdocs-old-versions.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+This may be because the new ``mkdocs==0.17.3`` version installed by Read the Docs introduced some breaking changes on the structure of the theme.
+
+You should check the `mkdocs' Custom Theme documentation`_ to upgrade your custom theme and make it compatible with the new version.
+
+.. _mkdocs' Custom Theme documentation: https://www.mkdocs.org/user-guide/custom-themes/
+
+
+Upgrade how extension arguments are defined
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
Comment (davidfischer) [inline docs/guides/mkdocs-old-versions.rst]: I actually couldn't find exactly where this syntax changed. I tested building docs with the old method in 0.15 and they still didn't build. Maybe this was a change in a markdown parser? I'm not sure.

Comment (humitos) [reply docs/guides/mkdocs-old-versions.rst]: Did you pin `Markdown` also to some version? I didn't, and I had to upgrade this.

**Comment by humitos** [comment]
Good points!

> * We should probably strongly recommend that people pin versions of mkdocs. The same problem may arise when we upgrade from 0.17 -> 1.x.

What do you think about adding a warning note in `docs/builds.rst::How to build documentation::MkDocs` like this:

```rst
.. warning::

   We strongly recommend to :ref:`pin the MkDocs version <guides/specifying-dependencies:Specifying Dependencies>`
   used for your project to build the docs to avoid potential future incompatibilities.
```

This way, if the user select MkDocs for his docs we are warning and pointing him to the right section about how to specify a dependency (which in the near future will be able from the YAML without needing a requirements.txt file).

On the other hand, supposing that we would have had this note in that place, the error would have happened again since in this particular case we need to pin other packages also :/. This note, will be helpful under "some" circuntances only.

> * This guide has time-sensitive information. In a year, this guide will probably not be needed and in 2 years it definitely won't be. What's the plan for deprecating and removing it?

I don't really have a plan :(

It's hard to remove the time-sensitive information since the user does know anything about the MkDocs version installed right now (because it's done automatically by RTD). So, talking about "Migrating from 0.15 version to X.X" or similar doesn't make too much sense to me. They don't really know that they strictly depend on `0.15`.

I was thinking on this guide like a temporal solution to point users if they open issues around this and/or to communicate via email to our .com customers.

This guide could probably completely removed some months after we notify our users/customers I'd say since it will become useless.

> * We should crosslink to the docs for specifying requirements

I added a new commit for this.

File: docs/guides/mkdocs-old-versions.rst (hunk 4 above)
Comment (davidfischer) [reply docs/guides/mkdocs-old-versions.rst]: Got it. Maybe this changed underneath mkdocs.

**Comment by davidfischer** [comment]
> What do you think about adding a warning note

Sounds good to me.

**Pull request merged**